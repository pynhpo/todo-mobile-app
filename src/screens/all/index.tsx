import React, {useState, useEffect, useMemo, useCallback} from 'react';
import {
  Text,
  View,
  Incubator,
  TouchableOpacity,
  Colors,
} from 'react-native-ui-lib';
import {SectionList, RefreshControl} from 'react-native';
import debounce from 'lodash/debounce';
import {useDispatch, useSelector} from 'react-redux';
import {styles} from './styles';
import {Layout} from '@components/layout.component';
import {SearchSvg} from '@components/svg/search-svg';
import {FilterSvg} from '@components/svg/filter-svg';
import {BluePlusSvg} from '@components/svg/blue-plus-svg';
import {NavigationService} from '@services/navigation.service';
import {SCREEN_NAME} from '@constants/navigation.constant';
import {fetchAllTodoAction} from '@redux/todo/all/actions';
import {setTitleOfTodoFilterState} from '@redux/todo/filter/slice';
import {initialTodoFilterState} from '@redux/todo/filter/constants';
import {selectAllTodo, selectTodoFilter} from '@redux/todo/selectors';
import {TodoItemResponse} from '@type/api';
import {TodoItem} from '@components/todo-item.component';

import {Filter} from './components/Filter';

const {TextField} = Incubator;

export const AllScreen = (): React.ReactElement => {
  const allTodo = useSelector(selectAllTodo);
  const filter = useSelector(selectTodoFilter);
  const dispatch = useDispatch();
  const [searchText, setSearchText] = useState('');
  const [isFilterVisible, setFilterVisible] = useState(false);

  const showFilter = () => {
    setFilterVisible(true);
  };

  const hideFilter = () => {
    setFilterVisible(false);
  };

  // eslint-disable-next-line react-hooks/exhaustive-deps
  const onSearchByTitle = useCallback(
    debounce((text: string) => {
      dispatch(setTitleOfTodoFilterState({title: text}));
    }, 400),
    [],
  );

  const onChangeSearchText = (text: string) => {
    setSearchText(text);
    onSearchByTitle(text);
  };

  const onRefresh = useCallback(() => {
    dispatch(fetchAllTodoAction());
  }, [dispatch]);

  useEffect(() => {
    onRefresh();
  }, [filter, onRefresh]);

  const isFiltering = useMemo(() => {
    if (
      filter.sortByName === initialTodoFilterState.sortByName &&
      filter.sortByOrder === initialTodoFilterState.sortByOrder &&
      !filter.status
    ) {
      return false;
    }
    return true;
  }, [filter]);

  const data = useMemo<
    Array<{
      title: string;
      data: TodoItemResponse[];
    }>
  >(() => {
    return Object.keys(allTodo.data.list).map(key => ({
      title: new Date(Number(key)).toLocaleDateString(),
      data: allTodo.data.list[key],
    }));
  }, [allTodo]);

  return (
    <Layout style={styles.root}>
      <View row center marginB-16>
        <TextField
          returnKeyType={'search'}
          placeholder="search title"
          grey10
          value={searchText}
          onChangeText={onChangeSearchText}
          autoCapitalize="none"
          leadingAccessory={
            <SearchSvg width={18} height={18} style={styles.searchIcon} />
          }
          containerStyle={styles.searchInput}
        />
        <TouchableOpacity onPress={showFilter} style={styles.filterIconCover}>
          <FilterSvg
            width={18}
            height={18}
            fill={Colors.gray}
            style={styles.filterIcon}
          />
          {isFiltering ? (
            <View
              absR
              width={7}
              height={7}
              br20
              backgroundColor={Colors.red30}
            />
          ) : null}
        </TouchableOpacity>
      </View>
      <SectionList
        sections={data}
        keyExtractor={item => item?.uid}
        renderItem={({item}) => <TodoItem item={item} />}
        renderSectionHeader={({section: {title}}) => (
          <View style={styles.sectionHeader} center>
            <Text text90 grey50>
              {title}
            </Text>
          </View>
        )}
        SectionSeparatorComponent={() => <View marginB-16 />}
        refreshControl={
          <RefreshControl
            refreshing={allTodo.isFetching}
            onRefresh={onRefresh}
          />
        }
        showsVerticalScrollIndicator={false}
      />
      <TouchableOpacity
        style={styles.bluePlusIconCover}
        onPress={() => {
          NavigationService.navigate(SCREEN_NAME.details);
        }}>
        <BluePlusSvg width={48} height={48} />
      </TouchableOpacity>
      <Filter isVisible={isFilterVisible} onHide={hideFilter} />
    </Layout>
  );
};
