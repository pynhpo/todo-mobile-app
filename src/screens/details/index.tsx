import React, {useState, FC, useCallback} from 'react';
import {KeyboardAvoidingScrollView} from '@components/keyboard-avoiding-scroll-view.component';
import {launchImageLibrary} from 'react-native-image-picker';
import storage from '@react-native-firebase/storage';
import map from 'lodash/map';
import {
  Text,
  View,
  Button,
  Incubator,
  TouchableOpacity,
  DateTimePicker,
  Picker,
  AnimatedImage,
  Colors,
} from 'react-native-ui-lib';
import {Platform, ActivityIndicator, ViewStyle} from 'react-native';
import {useDispatch} from 'react-redux';
import {styles} from './styles';
import {Layout} from '@components/layout.component';
import {ImageUploadSvg} from '@components/svg/image-upload-svg';
import {NavigationService} from '@services/navigation.service';
import {setOverlayLoadingModalState} from '@redux/modal/overlayLoading/slice';
import {
  addNewTodoItemAction,
  updateTodoItemAction,
} from '@redux/todo/item/actions';
import {Repeat, Priority, TodoItemResponse} from '@type/api';
import {Route} from '@type/navigation';
import {GapLine} from '@components/gap-line.component';

const repeatOptions: Array<{label: string; value: Repeat}> = [
  {label: 'None', value: 'none'},
  {label: 'Everyday', value: 'everyday'},
];

const priorityOptions: Array<{label: string; value: Priority}> = [
  {label: 'Normal', value: 'normal'},
  {label: 'Medium', value: 'medium'},
  {label: 'High', value: 'high'},
  {label: 'Highest', value: 'highest'},
];

const {TextField} = Incubator;

interface DetailsScreenProps
  extends Route<TodoItemResponse & {isToday?: boolean}> {}

export const DetailsScreen: FC<DetailsScreenProps> = (
  props,
): React.ReactElement => {
  const params = props?.route?.params;
  const isEditing = !!params?.uid;
  const dispatch = useDispatch();
  const [title, setTitle] = useState(params?.title || '');
  const [content, setContent] = useState(params?.content || '');
  const [appliedDate, setAppliedDate] = useState<Date>(
    params?.appliedAt ? new Date(params?.appliedAt) : new Date(),
  );
  const [dueDate, setDueDate] = useState<Date | undefined>(
    params?.dueAt ? new Date(params?.dueAt) : undefined,
  );
  const [dueTime, setDueTime] = useState<Date | undefined>(
    params?.dueAt ? new Date(params?.dueAt) : undefined,
  );
  const [repeat, setRepeat] = useState<Repeat>(
    params?.repeat || repeatOptions[0].value,
  );
  const [priority, setPriority] = useState<Priority>(
    params?.priority || priorityOptions[0].value,
  );
  const [imageUrl, setImageUrl] = useState<string>(params?.imageUrl || '');

  const onSave = () => {
    if (!title) {
      return;
    }
    const oneDayInMillisecond = 24 * 60 * 60 * 1000;
    const dueAt = dueDate
      ? Math.floor((dueDate?.getTime() || 0) / oneDayInMillisecond) *
          oneDayInMillisecond +
        (dueTime ? (dueTime?.getTime() || 0) % oneDayInMillisecond : 0)
      : undefined;

    const body = {
      title,
      content,
      appliedAt: appliedDate?.getTime(),
      dueAt,
      imageUrl,
      priority,
      repeat,
    };
    dispatch(
      isEditing
        ? updateTodoItemAction({...body, uid: params?.uid})
        : addNewTodoItemAction(body),
    );
  };

  const onLaunchImageLibrary = async () => {
    try {
      dispatch(setOverlayLoadingModalState({visible: true}));
      const result = await launchImageLibrary({
        mediaType: 'photo',
        quality: 0.5,
      });
      const uri = result.assets?.[0]?.uri;
      if (uri) {
        const filename = uri.substring(uri.lastIndexOf('/') + 1);
        const uploadUri =
          Platform.OS === 'ios' ? uri.replace('file://', '') : uri;
        await storage().ref(filename).putFile(uploadUri);
        const downloadURL = await storage().ref(filename).getDownloadURL();
        setImageUrl(downloadURL);
        dispatch(setOverlayLoadingModalState({visible: false}));
      }
    } catch (err) {
      dispatch(setOverlayLoadingModalState({visible: false}));
    }
  };

  const renderResetIcon = useCallback(
    (onPress = () => {}, style: ViewStyle) => {
      return (
        <TouchableOpacity style={style} onPress={onPress}>
          <View
            width={20}
            height={20}
            br100
            backgroundColor={Colors.red20}
            center>
            <Text text90 white>
              x
            </Text>
          </View>
        </TouchableOpacity>
      );
    },
    [],
  );

  const onResetImageUrl = () => {
    setImageUrl('');
  };

  const onResetDueDate = () => {
    setDueDate(undefined);
    setDueTime(undefined);
  };

  const onResetDueTime = () => {
    setDueTime(undefined);
  };

  return (
    <Layout useSafeAreaView style={styles.root}>
      <View row spread centerV marginB-16>
        <Button
          text80
          link
          label="Cancel"
          color={Colors.blue20}
          onPress={NavigationService.goBack}
        />
        <Text text70 grey10 marginR-8>
          Details
        </Text>
        <Button
          text80
          link
          label={isEditing ? 'Save' : 'Add'}
          onPress={onSave}
          color={Colors.blue20}
          disabled={!title}
        />
      </View>
      <KeyboardAvoidingScrollView>
        <View style={styles.mainContentCover}>
          <TextField
            placeholder="title"
            grey10
            value={title}
            onChangeText={setTitle}
            floatingPlaceholder
          />
          <GapLine marginT-8 />
          <TextField
            placeholder="content"
            grey10
            value={content}
            onChangeText={setContent}
            floatingPlaceholder
            multiline
          />
        </View>
        <View style={styles.dateCover}>
          {!params?.isToday ? (
            <DateTimePicker
              grey10
              text-90
              title="Applied date"
              placeholder="Press to choose"
              mode="date"
              onChange={(date: Date) => {
                setAppliedDate(date);
              }}
              value={appliedDate}
            />
          ) : null}
          <View style={styles.relative}>
            <DateTimePicker
              title="Due date"
              placeholder="Press to choose"
              mode="date"
              onChange={(date: Date) => setDueDate(date)}
              value={dueDate}
            />
            {dueDate
              ? renderResetIcon(onResetDueDate, styles.resetIconOfDueDate)
              : null}
          </View>

          <View style={styles.relative}>
            <DateTimePicker
              title="Due time"
              placeholder="Press to choose"
              mode="time"
              onChange={(date: Date) => setDueTime(date)}
              value={dueTime}
            />
            {dueTime
              ? renderResetIcon(onResetDueTime, styles.resetIconOfDueDate)
              : null}
          </View>
        </View>
        <View style={styles.repeatCover}>
          <Picker
            title="Repeat"
            placeholder="Pick a item"
            useNativePicker
            enableModalBlur
            value={repeat}
            onChange={(value: any) => setRepeat(value)}>
            {map(repeatOptions, option => (
              <Picker.Item
                key={option.value}
                value={option.value}
                label={option.label}
              />
            ))}
          </Picker>

          <Picker
            title="Priority"
            placeholder="Pick a item"
            enableModalBlur
            useNativePicker
            value={priority}
            onChange={(value: any) => {
              setPriority(value);
            }}>
            {map(priorityOptions, option => (
              <Picker.Item
                key={option.value}
                value={option.value}
                label={option.label}
              />
            ))}
          </Picker>
        </View>

        <TouchableOpacity onPress={onLaunchImageLibrary}>
          {!imageUrl ? (
            <View style={styles.imageUploadPlaceholder}>
              <ImageUploadSvg width={48} height={48} />
            </View>
          ) : (
            <View style={styles.relative}>
              <AnimatedImage
                style={styles.animatedImage}
                containerStyle={styles.animatedImageContainer}
                source={{uri: imageUrl}}
                loader={<ActivityIndicator color={Colors.primary} />}
                animationDuration={300}
              />
              {renderResetIcon(onResetImageUrl, styles.resetIconOfImage)}
            </View>
          )}
        </TouchableOpacity>
      </KeyboardAvoidingScrollView>
    </Layout>
  );
};
