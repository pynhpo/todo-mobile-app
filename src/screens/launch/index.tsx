import {View} from 'react-native';
import {logoutAction} from '@redux/auth/actions';
import {AppStorage} from '@services/app-storage.service';
import React, {useEffect} from 'react';
import {useDispatch} from 'react-redux';
import {NavigationService} from '@services/navigation.service';
import {CrashlyticsService} from '@services/crashlytics.service';
import {SCREEN_NAME} from '@constants/navigation.constant';
import {useSelector} from 'react-redux';
import {selectNavigationReady} from '@redux/app/selectors';

export const LaunchScreen = (): React.ReactElement => {
  const dispatch = useDispatch();
  const isReady = useSelector(selectNavigationReady);

  useEffect(() => {
    try {
      if (isReady) {
        const authToken = AppStorage.getAuthToken();
        if (authToken) {
          NavigationService.reset(SCREEN_NAME.mainTab);
        } else {
          NavigationService.reset(SCREEN_NAME.login);
        }
      }
    } catch (err) {
      dispatch(logoutAction());
      CrashlyticsService.recordError(err);
    }
  }, [dispatch, isReady]);
  return <View />;
};
