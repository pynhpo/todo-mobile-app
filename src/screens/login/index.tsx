import React, {useRef} from 'react';
import {KeyboardAvoidingScrollView} from '@components/keyboard-avoiding-scroll-view.component';
import {loginAction} from '@redux/auth/actions';
import {Text, View, TextField, Button} from 'react-native-ui-lib';
import {TextInput} from 'react-native';
import {useDispatch} from 'react-redux';
import {styles} from './styles';
import {Controller, useForm, SubmitHandler} from 'react-hook-form';
import {yupResolver} from '@hookform/resolvers/yup';
import {loginSchema} from '@constants/validation-schema.constant';
import {Layout} from '@components/layout.component';

type FormValues = {
  email: string;
  password: string;
};

export const LoginScreen = (): React.ReactElement => {
  const dispatch = useDispatch();
  const passwordInput =
    useRef<TextInput>() as React.MutableRefObject<TextInput>;

  const {handleSubmit, control} = useForm<FormValues>({
    resolver: yupResolver(loginSchema),
  });

  const onSubmit: SubmitHandler<FormValues> = data => {
    dispatch(loginAction({email: data.email, password: data.password}));
  };

  return (
    <Layout useSafeAreaView style={styles.safeArea}>
      <KeyboardAvoidingScrollView>
        <View flex paddingH-25 paddingT-120>
          <Text blue50 text20 marginB-16>
            Welcome
          </Text>
          <Controller
            control={control}
            render={({fieldState, field}) => (
              <TextField
                returnKeyType={'next'}
                text50
                placeholder="email"
                grey10
                value={field.value}
                onBlur={field.onBlur}
                onChangeText={field.onChange}
                onSubmitEditing={() => {
                  passwordInput && passwordInput.current.focus();
                }}
                validate={() => fieldState?.error}
                validationMessage={fieldState?.error?.message}
                autoCapitalize="none"
                enableErrors
              />
            )}
            name="email"
          />
          <Controller
            control={control}
            render={({fieldState, field}) => (
              <TextField
                autoCapitalize="none"
                text50
                placeholder="password"
                ref={passwordInput}
                secureTextEntry
                grey10
                value={field.value}
                onChangeText={field.onChange}
                onSubmitEditing={handleSubmit(onSubmit)}
                validate={() => fieldState?.error}
                validationMessage={fieldState?.error?.message}
                enableErrors
              />
            )}
            name="password"
          />

          <View marginT-100 center>
            <Button
              text70
              white
              background-orange30
              label="Login"
              onPress={handleSubmit(onSubmit)}
            />
          </View>
        </View>
      </KeyboardAvoidingScrollView>
    </Layout>
  );
};
