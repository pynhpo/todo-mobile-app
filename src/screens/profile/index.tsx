import React from 'react';
import {KeyboardAvoidingScrollView} from '@components/keyboard-avoiding-scroll-view.component';
import {Text, View, Button} from 'react-native-ui-lib';
import {useDispatch, useSelector} from 'react-redux';
import {styles} from './styles';
import {Layout} from '@components/layout.component';
import {selectUserEmail} from '@redux/auth/selectors';
import {logoutAction} from '@redux/auth/actions';

export const ProfileScreen = (): React.ReactElement => {
  const userEmail = useSelector(selectUserEmail);
  const dispatch = useDispatch();
  return (
    <Layout useSafeAreaView style={styles.safeArea}>
      <KeyboardAvoidingScrollView>
        <View flex paddingH-25 paddingT-120>
          <Text blue50 text20 marginB-16>
            {`Hi ${userEmail}`}
          </Text>
          <View marginT-100 center>
            <Button
              text70
              white
              background-orange30
              label="Logout"
              onPress={() => {
                dispatch(logoutAction());
              }}
            />
          </View>
        </View>
      </KeyboardAvoidingScrollView>
    </Layout>
  );
};
