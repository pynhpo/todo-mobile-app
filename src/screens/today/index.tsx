import React, {useEffect, useMemo} from 'react';
import {KeyboardAvoidingScrollView} from '@components/keyboard-avoiding-scroll-view.component';
import {
  Text,
  Incubator,
  TouchableOpacity,
  ListItem,
  Colors,
  Checkbox,
} from 'react-native-ui-lib';
import {SectionList} from 'react-native';
import {useDispatch, useSelector} from 'react-redux';
import {styles} from './styles';
import {Layout} from '@components/layout.component';
import {GreenPlusSvg} from '@components/svg/green-plus-svg';
import {NavigationService} from '@services/navigation.service';
import {SCREEN_NAME} from '@constants/navigation.constant';
import {fetchTodayTodoAction} from '@redux/todo/today/actions';
import {
  markTodoItemAsCompletedAction,
  deleteTodoItemAction,
} from '@redux/todo/item/actions';
import {selectTodayTodo} from '@redux/todo/selectors';
import {TodoItemResponse} from '@type/api';
import SwipeableAction from '@components/swipeable-actions.component';

export const TodayScreen = (): React.ReactElement => {
  const todayTodo = useSelector(selectTodayTodo);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchTodayTodoAction());
  }, [dispatch]);

  const data = useMemo<
    Array<{
      title: string;
      data: TodoItemResponse[];
    }>
  >(() => {
    return Object.keys(todayTodo.data.list).map(key => ({
      title: new Date(Number(key)).toLocaleDateString(),
      data: todayTodo.data.list[key],
    }));
  }, [todayTodo]);
  const updateItem = (item: TodoItemResponse) => {
    NavigationService.navigate(SCREEN_NAME.details, item);
  };
  return (
    <Layout style={styles.root}>
      <KeyboardAvoidingScrollView>
        <SectionList
          contentContainerStyle={{flex: 1}}
          sections={data}
          keyExtractor={item => item?.uid}
          renderItem={({item}) => (
            <SwipeableAction
              onEdit={() => {
                updateItem(item);
              }}
              onDelete={() => {
                dispatch(deleteTodoItemAction({uid: item?.uid}));
              }}>
              <ListItem
                activeBackgroundColor={Colors.grey60}
                activeOpacity={0.3}
                height={77.5}
                onPress={() => {
                  updateItem(item);
                }}>
                <ListItem.Part left>
                  {/* <Image source={{uri: row.mediaUrl}} style={styles.image} /> */}
                  <Checkbox
                    value={item.completed}
                    onValueChange={() => {
                      dispatch(
                        markTodoItemAsCompletedAction({
                          uid: item?.uid,
                          completed: !item.completed,
                        }),
                      );
                    }}
                  />
                </ListItem.Part>
                <ListItem.Part
                  middle
                  column
                  containerStyle={[styles.border, {paddingRight: 17}]}>
                  <ListItem.Part containerStyle={{marginBottom: 3}}>
                    <Text
                      grey10
                      text70
                      style={{flex: 1, marginRight: 10}}
                      numberOfLines={1}>
                      {item.title}
                    </Text>
                    <Text grey10 text70 style={{marginTop: 2}}>
                      {item.content}
                    </Text>
                  </ListItem.Part>
                  {/* <ListItem.Part>
                  <Text
                    style={{flex: 1, marginRight: 10}}
                    text90
                    grey40
                    numberOfLines={1}>{`${row.inventory.quantity} item`}</Text>
                  <Text text90 color={statusColor} numberOfLines={1}>
                    {row.inventory.status}
                  </Text>
                </ListItem.Part> */}
                </ListItem.Part>
              </ListItem>
            </SwipeableAction>
          )}
          renderSectionHeader={({section: {title}}) => <Text></Text>}
        />
      </KeyboardAvoidingScrollView>
      <TouchableOpacity
        style={styles.greenPlusIconCover}
        onPress={() => {
          NavigationService.navigate(SCREEN_NAME.details);
        }}>
        <GreenPlusSvg width={48} height={48} />
      </TouchableOpacity>
    </Layout>
  );
};
