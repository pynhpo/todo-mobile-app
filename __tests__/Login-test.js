import 'react-native';
import React from 'react';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';

import {LoginScreen} from '../src/screens/login';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('Login snapshot', () => {
  const initialState = {};
  const mockStore = configureStore();
  const store = mockStore(initialState);
  const snap = renderer
    .create(
      <Provider store={store}>
        <LoginScreen />
      </Provider>,
    )
    .toJSON();
  expect(snap).toMatchSnapshot();
});
