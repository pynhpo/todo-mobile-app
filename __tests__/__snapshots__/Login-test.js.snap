// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login snapshot 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "bottom",
      "left",
      "right",
      "top",
    ]
  }
  style={
    Array [
      Object {
        "backgroundColor": undefined,
      },
      Object {
        "flex": 1,
      },
    ]
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <RCTScrollView
      bounces={false}
      contentContainerStyle={Object {}}
      keyboardShouldPersistTaps="handled"
      showsHorizontalScrollIndicator={false}
      showsVerticalScrollIndicator={false}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View>
        <View
          paddingH-25={true}
          paddingT-100={true}
          style={
            Array [
              undefined,
              undefined,
              Object {
                "flex": 1,
              },
              undefined,
              Object {
                "paddingHorizontal": 25,
                "paddingTop": 100,
              },
              Object {},
              Object {},
              undefined,
            ]
          }
        >
          <Text
            blue50={true}
            marginB-16={true}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#20303C",
                  "textAlign": "left",
                },
                Object {
                  "fontFamily": "System",
                  "fontSize": 48,
                  "fontWeight": "400",
                  "lineHeight": 60,
                },
                Object {
                  "color": "#81B4FF",
                },
                undefined,
                undefined,
                Object {
                  "marginBottom": 16,
                },
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
            text20={true}
          >
            Welcome
          </Text>
          <View
            style={
              Array [
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {},
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            <Text
              color="#20303C"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#20303C",
                    "textAlign": "left",
                  },
                  undefined,
                  Object {
                    "color": "#20303C",
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "minHeight": 20,
                    },
                    undefined,
                    Object {
                      "opacity": 0,
                    },
                  ],
                ]
              }
              testID="undefined.label"
            />
            <View
              centerV={true}
              row={true}
              style={
                Array [
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Object {},
                  Object {},
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  Array [
                    Object {},
                    Array [
                      undefined,
                      undefined,
                    ],
                  ],
                ]
              }
            >
              <View
                flexG={true}
                style={
                  Array [
                    undefined,
                    undefined,
                    Object {
                      "flexGrow": 1,
                    },
                    undefined,
                    Object {},
                    Object {},
                    Object {},
                    undefined,
                  ]
                }
              >
                <View
                  absF={true}
                  style={
                    Array [
                      undefined,
                      undefined,
                      undefined,
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {},
                      Object {},
                      Object {},
                      false,
                    ]
                  }
                >
                  <Text
                    animated={true}
                    collapsable={false}
                    color="#A6ACB1"
                    onLayout={[Function]}
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#A6ACB1",
                        "fontFamily": "System",
                        "fontSize": 20,
                        "fontWeight": "800",
                        "textAlign": "left",
                        "transform": Array [
                          Object {
                            "scale": 0.875,
                          },
                          Object {
                            "translateX": 0,
                          },
                          Object {
                            "translateY": 0,
                          },
                        ],
                      }
                    }
                    testID="undefined.floatingPlaceholder"
                  >
                    email
                  </Text>
                </View>
                <TextInput
                  accessibilityState={
                    Object {
                      "disabled": false,
                    }
                  }
                  autoCapitalize="none"
                  grey20={true}
                  isValid={true}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  onSubmitEditing={[Function]}
                  placeholder="email"
                  placeholderTextColor="transparent"
                  returnKeyType="next"
                  style={
                    Array [
                      Object {
                        "flexGrow": 1,
                        "paddingBottom": 0,
                        "paddingTop": 0,
                        "textAlign": "left",
                      },
                      Object {
                        "color": "#20303C",
                      },
                      Array [
                        Object {
                          "fontFamily": "System",
                          "fontSize": 20,
                          "fontWeight": "800",
                        },
                        Object {
                          "color": "#4D5963",
                        },
                        undefined,
                      ],
                    ]
                  }
                  text60={true}
                  underlineColorAndroid="transparent"
                  value="username_1@todo.com"
                />
              </View>
            </View>
            <View
              row={true}
              spread={true}
              style={
                Array [
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Object {},
                  Object {},
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                  },
                  undefined,
                ]
              }
            >
              <Text
                $textDangerLight={true}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#20303C",
                      "textAlign": "left",
                    },
                    undefined,
                    Object {
                      "color": "#FC3D2F",
                    },
                    undefined,
                    undefined,
                    Object {},
                    undefined,
                    undefined,
                    undefined,
                    Array [
                      Object {
                        "minHeight": 20,
                      },
                      undefined,
                    ],
                  ]
                }
                testID="undefined.validationMessage"
              />
            </View>
          </View>
          <View
            marginB-20={true}
            style={
              Array [
                undefined,
                undefined,
                undefined,
                undefined,
                Object {},
                Object {
                  "marginBottom": 20,
                },
                Object {},
                Array [
                  Object {
                    "backgroundColor": undefined,
                    "height": 1,
                    "width": "100%",
                  },
                  undefined,
                ],
              ]
            }
          />
          <View
            style={
              Array [
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {},
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            <View
              centerV={true}
              row={true}
              style={
                Array [
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Object {},
                  Object {},
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  Array [
                    Object {},
                    Array [
                      undefined,
                      undefined,
                    ],
                  ],
                ]
              }
            >
              <View
                flexG={true}
                style={
                  Array [
                    undefined,
                    undefined,
                    Object {
                      "flexGrow": 1,
                    },
                    undefined,
                    Object {},
                    Object {},
                    Object {},
                    undefined,
                  ]
                }
              >
                <TextInput
                  accessibilityState={
                    Object {
                      "disabled": false,
                    }
                  }
                  autoCapitalize="none"
                  grey20={true}
                  isValid={true}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  onSubmitEditing={[Function]}
                  placeholder="password"
                  secureTextEntry={true}
                  style={
                    Array [
                      Object {
                        "flexGrow": 1,
                        "paddingBottom": 0,
                        "paddingTop": 0,
                        "textAlign": "left",
                      },
                      Object {
                        "color": "#20303C",
                      },
                      Array [
                        Object {
                          "fontFamily": "System",
                          "fontSize": 20,
                          "fontWeight": "800",
                        },
                        Object {
                          "color": "#4D5963",
                        },
                        undefined,
                      ],
                    ]
                  }
                  text60={true}
                  underlineColorAndroid="transparent"
                  value="123456"
                />
              </View>
            </View>
            <View
              row={true}
              spread={true}
              style={
                Array [
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Object {},
                  Object {},
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                  },
                  undefined,
                ]
              }
            >
              <Text
                $textDangerLight={true}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#20303C",
                      "textAlign": "left",
                    },
                    undefined,
                    Object {
                      "color": "#FC3D2F",
                    },
                    undefined,
                    undefined,
                    Object {},
                    undefined,
                    undefined,
                    undefined,
                    Array [
                      Object {
                        "minHeight": 20,
                      },
                      undefined,
                    ],
                  ]
                }
                testID="undefined.validationMessage"
              />
            </View>
          </View>
          <View
            center={true}
            marginT-68={true}
            style={
              Array [
                undefined,
                undefined,
                undefined,
                undefined,
                Object {},
                Object {
                  "marginTop": 68,
                },
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <View
              accessibilityRole="button"
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onLayout={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#5A48F5",
                  "borderRadius": 999,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "minWidth": 90,
                  "opacity": 1,
                  "paddingHorizontal": 20,
                  "paddingVertical": 9.5,
                }
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#20303C",
                      "textAlign": "left",
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "flex": 0,
                        "flexDirection": "row",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": "400",
                        "lineHeight": 24,
                      },
                      Object {
                        "color": "#FFFFFF",
                      },
                      undefined,
                      Object {
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": "400",
                        "lineHeight": 24,
                      },
                      undefined,
                    ],
                  ]
                }
                testID="undefined.label"
              >
                Login
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RNCSafeAreaView>
`;
